docs/source/optimization.rst

```diff
@@ -0,0 +1,140 @@
+          scipy.optimize.minimize interface to L-BFGS-B.
+      maxfun : int
+          Maximum number of iterations.
+      epsilon : float
+          Step size used when approx_grad is True, for numerically
+          calculating the gradient
+      approx_grad : bool
+          Whether to approximate the gradient numerically (in which
+          case func returns only the function value).
+- ``cg`` - Conjugate gradient optimization. Scipy's ``fmin_cg``.
+      gtol : float
+          Stop when norm of gradient is less than gtol.
+      norm : float
+          Order of norm (np.Inf is max, -np.Inf is min)
+      epsilon : float
+          If fprime is approximated, use this value for the step
+          size. Can be scalar or vector.  Only relevant if
+          Likelihoodmodel.score is None.
+- ``ncg`` - Newton conjugate gradient. Scipy's ``fmin_ncg``.
+      fhess_p : callable f'(x, \*args)
+          Function which computes the Hessian of f times an arbitrary
+          vector, p.  Should only be supplied if
+          LikelihoodModel.hessian is None.
+      avextol : float
+          Stop when the average relative error in the minimizer
+          falls below this amount.
+      epsilon : float or ndarray
+          If fhess is approximated, use this value for the step size.
+          Only relevant if Likelihoodmodel.hessian is None.
+- ``powell`` - Powell's method. Scipy's ``fmin_powell``.
+      xtol : float
+          Line-search error tolerance
+      ftol : float
+          Relative error in loglike(params) for acceptable for
+          convergence.
+      maxfun : int
+          Maximum number of function evaluations to make.
+      start_direc : ndarray
+          Initial direction set.
+- ``basinhopping`` - Basin hopping. This is part of scipy's ``basinhopping``
+  tools.
+      niter : integer
+          The number of basin hopping iterations.
+      niter_success : integer
+          Stop the run if the global minimum candidate remains the
+          same for this number of iterations.
+      T : float
+          The "temperature" parameter for the accept or reject
+          criterion. Higher "temperatures" mean that larger jumps
+          in function value will be accepted. For best results
+          `T` should be comparable to the separation (in function
+          value) between local minima.
+      stepsize : float
+          Initial step size for use in the random displacement.
+      interval : integer
+          The interval for how often to update the `stepsize`.
+      minimizer : dict
+          Extra keyword arguments to be passed to the minimizer
+          `scipy.optimize.minimize()`, for example 'method' - the
+          minimization method (e.g. 'L-BFGS-B'), or 'tol' - the
+          tolerance for termination. Other arguments are mapped from
+          explicit argument of `fit`:
+          - `args` <- `fargs`
+          - `jac` <- `score`
+          - `hess` <- `hess`
+
```

Comment:
this is missing our newer generic interface to scipy `minimize` which allows to use any of the scipy optimization algorithms.

Reply:
I saw that, but I honestly have never experimented with it or even tested it. Is there a simple example for this? It sounds useful.